WebAuthn docs update (#25922)
mdn/content: files/en-us/web/api/web_authentication_api/index.md

4. The web app forwards the {{domxref("PublicKeyCredential")}} to the server, again using an appropriate mechanism.

5. The server stores the public key, coupled with the user identity, to remember the credential for future authentications. During this process, it performs a series of checks to ensure that the registration was complete and not tampered with. These include:

   1. Verifying that the challenge is the same as the challenge that was sent.
   2. Ensuring that the origin was the origin expected.
   3. Validating that the signature and attestation are using the correct certificate chain for the specific model of the authenticator used to generated the key par in the first place.

> **Warning:** Attestation provides a way for a relying party to determine the provenance of an authenticator. Relying parties should not attempt to maintain allowlists of authenticators.

### Authenticating a user

> After a user has registered with WebAuthn, they can subsequently authenticate (i.e. login) with the service. The authentication flow looks similar to the registration flow, the main differences being that authentication:

Reviewer: ```suggestion
After a user has registered with WebAuthn, they can authenticate (i.e., login) with the service. The authentication flow looks similar to the registration flow, the main differences being that authentication:
```

If you wish to authenticate with `get()` in an `<iframe>`, there are a couple of steps to follow:

1. The site embedding the relying party site must provide permission via an `allow` attribute:

   ```html
   <iframe
     src="https://auth.provider.com"
     allow="publickey-credentials-get *" />
   ```

2. The relying party site must provide permission for the above access via a `Permissions-Policy` header:

   ```http
   Permissions-Policy: publickey-credentials-get=*
   ```

   Or to allow only a specific URL to embed the relying party site in an `<iframe>`:

   ```http
   Permissions-Policy: publickey-credentials-get=("https://subdomain.example.com")
   ```

## Interfaces

- {{domxref("AuthenticatorAssertionResponse")}}
  - : Provides proof to a service that an authenticator has the necessary key pair to successfully handle an authentication request initiated by a {{domxref("CredentialsContainer.get()")}} call. Available in the {{domxref("PublicKeyCredential.response", "response")}} property of the {{domxref("PublicKeyCredential")}} instance obtained when the `get()` {{jsxref("Promise")}} fulfills.
- {{domxref("AuthenticatorAttestationResponse")}}

>   - : is the result of a WebAuthn credential registration (i.e. a {{domxref("CredentialsContainer.create()")}} call). It contains information about the credential that the server needs to perform WebAuthn assertions, such as its credential ID and public key. Available in the {{domxref("PublicKeyCredential.response", "response")}} property of the {{domxref("PublicKeyCredential")}} instance obtained when the `create()` {{jsxref("Promise")}} fulfills.

Reviewer: ```suggestion
  - : The result of a WebAuthn credential registration (i.e., a {{domxref("CredentialsContainer.create()")}} call). It contains information about the credential that the server needs to perform WebAuthn assertions, such as its credential ID and public key. Available in the {{domxref("PublicKeyCredential.response", "response")}} property of the {{domxref("PublicKeyCredential")}} instance obtained when the `create()` {{jsxref("Promise")}} fulfills.
```

### Usage example

> > **Note:** For security reasons, WebAuthn calls ({{domxref("CredentialsContainer.create","create()")}} and {{domxref("CredentialsContainer.get","get()")}}) are cancelled if the browser window loses focus while the call is pending.

Reviewer: ```suggestion
> **Note:** For security reasons, the Web Authentication API calls ({{domxref("CredentialsContainer.create", "create()")}} and {{domxref("CredentialsContainer.get","get()")}}) are canceled if the browser window loses focus while the call is pending.
```
